src/libvirtApi/domain.js

```diff
@@ -761,11 +759,13 @@ export function domainSetVCPUSettings ({
     threads,
     isRunning
 }) {
-    return call(connectionName, objPath, 'org.libvirt.Domain', 'GetXMLDesc', [Enum.VIR_DOMAIN_XML_INACTIVE], { timeout, type: 'u' })
-            .then(domXml => {
-                const updatedXML = updateVCPUSettings(domXml[0], count, max, sockets, cores, threads);
-                return call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'DomainDefineXML', [updatedXML], { timeout, type: 's' });
-            });
+    const opts = { err: "message", environ: ['LC_ALL=C'] };
+    if (connectionName === 'system')
+        opts.superuser = 'try';
+
+    return cockpit.script(
+        `virt-xml -c qemu:///${connectionName} --vcpu ${max},vcpu.current=${count},sockets=${sockets},cores=${cores},threads=${threads} ${name} --edit`, opts
```

Comment:
Can we please not introduce shell again? We just got rid of this with converting the huge shell scripts to Python. If the name contains spaces, quotes, or any funny characters, this is going to blow up in the user's face. Let's please use cockpit.spawn() and assemble a proper array. This is not significantly more difficult, but dramatically more robust.

Reply:
Of course! Sorry for not realizing this!